src/esp/bindings/PhysicsBindings.cpp

```diff
@@ -69,6 +72,69 @@ void initPhysicsBindings(py::module& m) {
       .def_readwrite("linear_friction_direction2",
                      &ContactPointData::linearFrictionDirection2)
       .def_readwrite("is_active", &ContactPointData::isActive);
+
+  // ==== enum object CollisionGroup ====
+  py::enum_<CollisionGroup> collisionGroups{m, "CollisionGroups",
+                                            "CollisionGroups"};
+  collisionGroups.value("Default", CollisionGroup::Default)
+      .value("Static", CollisionGroup::Static)
+      .value("Kinematic", CollisionGroup::Kinematic)
+      .value("FreeObject", CollisionGroup::FreeObject)
+      .value("Robot", CollisionGroup::Robot)
+      .value("Noncollidable", CollisionGroup::Noncollidable)
+      .value("UserGroup0", CollisionGroup::UserGroup0)
+      .value("UserGroup1", CollisionGroup::UserGroup1)
+      .value("UserGroup2", CollisionGroup::UserGroup2)
+      .value("UserGroup3", CollisionGroup::UserGroup3)
+      .value("UserGroup4", CollisionGroup::UserGroup4)
+      .value("UserGroup5", CollisionGroup::UserGroup5)
+      .value("UserGroup6", CollisionGroup::UserGroup6)
+      .value("UserGroup7", CollisionGroup::UserGroup7)
+      .value("UserGroup8", CollisionGroup::UserGroup8)
+      .value("UserGroup9", CollisionGroup::UserGroup9)
+      .value("None", CollisionGroup{});
+  corrade::enumOperators(collisionGroups);
+
+  // ==== class object CollisionGroupHelper ====
+  py::class_<CollisionGroupHelper, std::shared_ptr<CollisionGroupHelper>>(
+      m, "CollisionGroupHelper")
+      .def_static("get_group", &CollisionGroupHelper::getGroup, "name"_a,
+                  R"(Get a group by assigned name.)")
+      .def_static("get_group_name", &CollisionGroupHelper::getGroupName,
+                  "group"_a, R"(Get the name assigned to a CollisionGroup.)")
+      .def_static("set_group_name", &CollisionGroupHelper::setGroupName,
+                  "group"_a, "name"_a, R"(Assign a name to a CollisionGroup.)")
+      .def_static(
+          "get_mask_for_group",
+          [](CollisionGroup group) {
+            return CollisionGroup(
+                uint32_t(CollisionGroupHelper::getMaskForGroup(group)));
+          },
+          "group"_a,
+          R"(Get the mask for a collision group describing its interaction with other groups.)")
+      .def_static(
+          "get_mask_for_group",
+          [](const std::string& group_name) {
+            return CollisionGroup(
+                uint32_t(CollisionGroupHelper::getMaskForGroup(group_name)));
+          },
+          "group"_a,
+          R"(Get the mask for a collision group describing its interaction with other groups.)")
+      .def_static(
+          "set_mask_for_group",
+          [](CollisionGroup group, CollisionGroup mask) {
```

Comment:
I am puzzled by `CollisionGroup mask`. Shouldn't the type of mask be `int` or `unsigned int`? When you do bitwise-or of a few CollisionGroup enum values, the resulting value is *not* a valid CollisionGroup enum value, right?

Reply:
I spoke to Mosra about this. Essentially the CollisionGroup enum variable here will hold the uint mask value on the python side without an explicit category name (e.g. as produced by the set operators) and this lambda casts that value to a C++ EnumSet at this layer. I know it looks a bit odd.